Yank/experiment.py

```diff
@@ -1479,7 +1575,167 @@ def _setup_experiments(self):
+            CA atoms constrained during optimization (default is True).
+        n_equilibration_iterations : None or int
+            The number of equilibration iterations to perform before running
+            the path search. If None, the function will determine the number
+            of iterations to run based on the system dimension.
+
+        Other Parameters
+        ----------------
+        kwargs : dict
+            Other parameters to pass to pipeline.find_alchemical_protocol().
+
+        """
+        class DummyReporter(object):
+            """A dummy reporter since we don't need to store repex stuff on disk."""
+            def nothing(self, *args, **kwargs):
+                """This serves both as an attribute and a callable."""
+                pass
+            def __getattr__(self, _):
+                return self.nothing
+
+        # Unpack experiment argument that has been distributed among nodes.
+        experiment_path, experiment = experiment
+
+        # Maybe only a subset of the phases need to be generated.
+        protocol = self._protocols[experiment['protocol']]
+        phases_to_generate = self._find_automatic_protocol_phases(protocol)
+
+        # Build experiment.
+        exp = self._build_experiment(experiment_path, experiment)
+
+        # Generate protocols.
+        optimal_protocols = collections.OrderedDict.fromkeys(phases_to_generate)
+        for phase_idx, phase_name in enumerate(phases_to_generate):
+            logger.debug('Generating alchemical path for {}.{}'.format(experiment_path, phase_name))
+            phase = exp.phases[phase_idx]
+            state_parameters = []
+            is_vacuum = len(phase.topography.receptor_atoms) == 0 and len(phase.topography.solvent_atoms) == 0
+
+            # We may need to slowly turn on a Boresch restraint.
+            if isinstance(phase.restraint, restraints.Boresch):
+                state_parameters.append(('lambda_restraints', [0.0, 1.0]))
+
+            # We support only lambda sterics and electrostatics for now.
+            if is_vacuum and not phase.alchemical_regions.annihilate_electrostatics:
+                state_parameters.append(('lambda_electrostatics', [1.0, 1.0]))
+            else:
+                state_parameters.append(('lambda_electrostatics', [1.0, 0.0]))
+            if is_vacuum and not phase.alchemical_regions.annihilate_sterics:
+                state_parameters.append(('lambda_sterics', [1.0, 1.0]))
+            else:
+                state_parameters.append(('lambda_sterics', [1.0, 0.0]))
+
+            # We only need to create a single state.
+            phase.protocol = {par[0]: [par[1][0]] for par in state_parameters}
+
+            # Remove unsampled state that we don't need for the optimization.
+            phase.options['anisotropic_dispersion_correction'] = False
+
+            # If default argument is used, determine number of equilibration iterations.
+            # TODO automatic equilibration?
```

Comment:
> I just object to the hard-coding of these values for now

Yeah, I understand, I don't like it either. The reason I did it is that everything is still very experimental, and I didn't want to expose YAML options in YANK 1.0 that we would have to break after few weeks.

If you agree, I'd leave it as it is now, and postpone this decision to after we have generated the first paths for the YANK 1.0 paper simulations, so that we'll have a better idea of what we actually need before engineering stuff in. We can use the API to make experiments.